python/podio/pythonizations/collection_subscript.py

```diff
@@ -1,8 +1,17 @@
 """Pythonize subscript operation for collections"""
 
+import warnings
 import cppyy
 from .utils.pythonizer import Pythonizer
 
+warnings.warn(
+    "The module 'collection_subscript' is deprecated and will be removed in a future release. "
+    "The subscript pythonization is now defined in generated C++ class definition "
+    "and applied automatically.",
+    FutureWarning,
+    stacklevel=2,
+)
+
```

Comment:
Thinking about it, `DeprecationWarning` might be more appropriate here. Sorry for being slow.

Just to be sure that I understand how things work **now**: We get the `collection_subscript` pythonization automatically (without possibility to opt out). Previously, you either get it via the catch-all `load_pythonizations` or via the explicit registration of just this pythonization.

Hence, people just using `load_pythonizations` have no change in behavior, but now see a`FutureWarning`, for which there is not even an easy fix, because for them the import happens "magically" behind `load_pythonizations`.